src/index.js

```diff
@@ -362,8 +362,13 @@ function fixResponseChunkedTransferBadEnding(request, errorCallback) {
 		const {headers} = response;
 		if (headers['transfer-encoding'] === 'chunked' && !headers['content-length']) {
 			response.once('close', hadError => {
-				// if a data listener is still present we didn't end cleanly
-				const hasDataListener = socket.listenerCount('data') > 0;
+				// tests for socket presence, as in some situations the
+				// the 'socket' event is not triggered for the request
+				// (happens in deno), avoids `TypeError`
+				if (socket !== undefined) {
+					// if a data listener is still present we didn't end cleanly
+					const hasDataListener = socket.listenerCount('data') > 0;
+				}
```

Comment:
This `const` declaration only exists within the if statements scope, and not outside it where it's later being used

```suggestion
                                let hasDataListener
				if (socket !== undefined) {
					// if a data listener is still present we didn't end cleanly
					hasDataListener = socket.listenerCount('data') > 0;
				}
```